src/Handler/StreamHandler.php

```diff
@@ -428,7 +430,24 @@ function ($code, $a, $b, $c, $transferred, $total) use ($value) {
         );
     }
 
-    private function add_debug(RequestInterface $request, &$options, $value, &$params)
+    private function add_force_ip_resolve(RequestInterface $request, &$options, $value, &$params, UriInterface &$uri)
+    {
+        if ('v4' === $value) {
+            $records = dns_get_record($uri->getHost(), DNS_A);
+            if (!isset($records[0]['ip'])) {
+                throw new ConnectException("Could not resolve IPv4 address for host '{$uri->getHost()}'", $request);
+            }
+            $uri = $uri->withHost($records[0]['ip']);
```

Comment:
Sadly this is also not going to work as you expect. HTTP Messages are immutable, which means if you "modify" one them you will get a new object. The new URI will not be present in the request object. What you can do is saving the resolved IP address in the options array and passing it to the URI, then the URI to the request somewhere in the above code.